lib/object/has-property.js

```diff
@@ -5,22 +5,41 @@
  * @author Eric Satterwhite
  **/
 
-module.exports = function hasOwnProperty(obj, prop) {
-  return Object.prototype.hasOwnProperty.call(obj, prop)
+module.exports = function hasProperty(obj, string = '', sep = '.') {
+  /* eslint-disable-next-line no-eq-null */
+  if (obj == null) return false
```

Comment:
I think rather than a default property on `string`, id let it be undefined.

```javascript
if (!string) return false
if (typeof string !== 'string') return false
```

Reply:
Done! Updated!